src/lib/pickles/compile.ml

```diff
@@ -563,44 +555,51 @@ struct
                 ~wrap_rounds:Tock.Rounds.n
 
             let f (T b : _ Branch_data.t) =
-              let (T (typ, _conv, conv_inv)) = etyp in
-              let main () () =
-                let res = b.main ~step_domains () in
-                Impls.Step.with_label "conv_inv" (fun () -> conv_inv res)
-              in
-              let () = if true then log_step main typ name b.index in
               let open Impls.Step in
               (* HACK: TODO docs *)
-              if return_early_digest_exception then
-                raise
-                  (Return_digest
-                     ( constraint_system ~input_typ:Typ.unit ~return_typ:typ main
-                     |> R1CS_constraint_system.digest ) ) ;
+              if return_early_digest_exception then failwith "TODO: Delete me" ;
```

Comment:
Not quite sure why we have that. Shouldn't we just ignore `return_early_digest_exception`?

Reply:
We should remove the argument althogether. If noone objects I can do it in this PR